src/Draggable/Sensors/DragSensor/tests/DragSensor.test.js

```diff
@@ -0,0 +1,514 @@
+ * drag and drop libraries and testing contexts as well. See below.
+ *
+ * @todo Find a way to test this behavior using the real objects
+ * @see https://reactjsnews.com/testing-drag-and-drop-components-in-react-js
+ * @see https://github.com/angular-dragdrop/angular-dragdrop/issues/22
+ *
+ * @return {Object}
+ */
+function getDataTransferStub() {
+  return {
+    setData: jest.fn(),
+    effectAllowed: null,
+  };
+}
+
+describe('DragSensor', () => {
+
+  describe('#constructor', () => {
+    test('should be instance of DragSensor', () => {
+      const dragSensor = new DragSensor();
+
+      expect(dragSensor).toBeInstanceOf(DragSensor);
+    });
+
+    test('should initialize dragging state to false', () => {
+      const dragSensor = new DragSensor();
+
+      expect(dragSensor.dragging).toBe(false);
+    });
+
+    test('should initialize the current container to null', () => {
+      const dragSensor = new DragSensor();
+
+      expect(dragSensor.currentContainer).toBe(null);
+    });
+  });
+
+  describe('#attach', () => {
+    test('should add event listeners to each container', () => {
+      const containers = [{}, {}, {}];
+
+      // Mock addEventListener for each container
+      containers.forEach((container) => {
+        container.addEventListener = jest.fn();
+      });
+
+      const dragSensor = new DragSensor(containers);
+
+      // Before calling attach, no events should have been added
+      containers.forEach((container) => {
+        expect(container.addEventListener.mock.calls.length).toBe(0);
+      });
+
+      dragSensor.attach();
+
+      // After calling attach, 5 events should have been added
+      containers.forEach((container) => {
+        expect(container.addEventListener.mock.calls.length).toBe(5);
+      });
+
+      // And the events should be correct
+      containers.forEach((container) => {
+        expect(container.addEventListener.mock.calls[0])
+          .toMatchObject(['mousedown', dragSensor._onMouseDown, true]);
+
+        expect(container.addEventListener.mock.calls[1])
+          .toMatchObject(['dragstart', dragSensor._onDragStart, false]);
+
+        expect(container.addEventListener.mock.calls[2])
+          .toMatchObject(['dragover', dragSensor._onDragOver, false]);
+
+        expect(container.addEventListener.mock.calls[3])
+          .toMatchObject(['dragend', dragSensor._onDragEnd, false]);
+
+        expect(container.addEventListener.mock.calls[4])
+          .toMatchObject(['drop', dragSensor._onDragDrop, false]);
+      });
+    });
+
+    test('should add mouseup event listener to document', (done) => {
+      const dragSensor = new DragSensor();
+
+      const originalAddEventListener = document.addEventListener;
+
+      document.addEventListener = jest.fn();
+
+      dragSensor.attach();
+
+      expect(document.addEventListener.mock.calls.length).toBe(1);
+
+      document.addEventListener = originalAddEventListener;
+
+      done();
```

Comment:
Not sure if I am missing something, but do we need the `done` callback? Don't see any async action